packages/components/calendar/src/use-month-picker.ts

```diff
@@ -111,11 +113,26 @@ export const useMonthPicker = ({
     onChangeType: (type, year, month) => {
       if (type !== "date") {
         setType(type)
+
+        if (year) {
+          setIsClickOnYearListAlready(true)
+
+          if (isClickOnYearListAlready) {
+            const prevMonth = value?.getMonth()
+
+            const currentValue: Date = new Date(year, prevMonth || 0)
+            const inputValue = dateToString(currentValue)
+
+            setValue(currentValue)
+            setInputValue(inputValue)
+          }
+        }
```

Comment:
Basiclly , we already have base value month , year when focus on Month picker Header .

So I wonder if this way is valid .

The main idea is that I will set value only after Button Year is clicked .